server/spec/derived_product_import_spec.rb

```diff
@@ -54,15 +54,20 @@
       {
         :derived_product_id => derived_product.id
       })
-    pool = @cp.list_owner_pools(@owner['key'], {:product => stacked_datacenter_product.id})[0]
+
+    pools = @cp.list_owner_pools(@owner['key'], {:product => stacked_datacenter_product.id})
+
+    pool  = pools.select do |p|
+      p['type'] == 'NORMAL'
```

Comment:
I believe this might be the wrong way to make the test pass. It's a test about derived products, and, looking at how the test is setting up it's data, its currently wrong. We no longer specify the derived product when creating the pool, but when we create the main (marketing) product. Thus, the order of data creation would be: 1) create the derived product, 2) create the main product and specify the derived product we created in step 1, 3) create the pool by specifying only the main product (and preferably by using the `create_pool()` method instead of the deprecated `create_pool_and_subscription`.

Reply:
Updated the code to create the products and pool in the correct order.